cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1642,6 +1642,55 @@ const char* convert_toUTF8(int64_t context, const char* value, int32_t value_len
   return value;
 }
 
+// Calculate the levenshtein distance between two string values
+FORCE_INLINE
+gdv_int32 levenshtein_utf8_utf8(int64_t context, const char* in1, int32_t in1_len,
+                                const char* in2, int32_t in2_len) {
+  if (in1_len < 0 || in2_len < 0) {
+    gdv_fn_context_set_error_msg(context, "String length must be greater than 0");
+    return 0;
+  }
+  int32_t a, b, c, aux;
+  int32_t len_dist_1 = in1_len + 1;
+  int32_t len_dist_2 = in2_len + 1;
+  // dist[i][j] represents the Levenstein distance between the strings
+  int** dist;
+  if ((dist = (int**)malloc((len_dist_1 + 1) * sizeof(int*))) == NULL) {
```

Comment:
Why len_dist_1 + 1 and not len_dist_1?

The algorithm in wikipedia allocates a matrix of size (in1_len + 1) x (in2_len + 1)

Reply:
Yeah, right, i removed this, thank you!